include/velocypack/Iterator.h

```diff
@@ -313,27 +299,61 @@ class ObjectIterator {
     return _slice.getNthValue(_position);
   }
 
-  inline void next() { operator++(); }
+  void next() noexcept {
+    if (VELOCYPACK_UNLIKELY(!valid())) {
+      return;
+    }
+    ++_position;
+    if (_current != nullptr) {
+      // skip over key
+      _current += Slice(_current).byteSize();
+      // skip over value
+      _current += Slice(_current).byteSize();
+    }
+  }
 
-  inline ValueLength index() const noexcept { return _position; }
+  [[nodiscard]] ValueLength index() const noexcept { return _position; }
 
-  inline ValueLength size() const noexcept { return _size; }
+  [[nodiscard]] ValueLength size() const noexcept { return _size; }
 
-  inline bool isFirst() const noexcept { return (_position == 0); }
+  [[deprecated(
+      "Dangerous API, you don't really know valid or not. "
+      "You can use index() and size().")]] [[nodiscard]] bool
+  isFirst() const noexcept {
+    return _position == 0;
+  }
 
-  inline bool isLast() const noexcept { return (_position + 1 >= _size); }
+  [[deprecated(
+      "Dangerous API, you don't really know valid or not. "
+      "You can use index() and size().")]] [[nodiscard]] bool
+  isLast() const noexcept {
+    return _position + 1 >= _size;
+  }
 
-  inline void reset() {
+  void reset() noexcept {
     _position = 0;
-    _current = _first;
+    _current = first(_current != nullptr);
   }
 
  private:
+  [[nodiscard]] uint8_t const* first(bool useSequentialIteration) noexcept {
+    if (VELOCYPACK_UNLIKELY(_size == 0)) {
+      return nullptr;
+    }
+    auto const head = _slice.head();
+    VELOCYPACK_ASSERT(head != 0x0a);  // no empty object allowed here
+    if (head == 0x14) {
```

Comment:
I guess it should rather be like it was in https://github.com/arangodb/velocypack/pull/141/files#diff-c1541b17db38faa9c3d35b5518644ce5f3ccdeeffc0a2de7ba65deb33e99bab6L73:
```suggestion
    VELOCYPACK_ASSERT(head != 0x01);  // no empty array allowed here
    if (head == 0x13) {
```

Reply:
Probably bad copy

Reply:
Fixed